Replace fact categories with tags

UtkarshKr007 opened the pull request:
We have been sorting facts into categories like 'habitat' and 'behavior', but not all facts can be easily categorized into a single criteria. A category like 'habitat' can contain facts about both 'plants' and 'homes' associated with pollinators which is why we are switching to tags. Tags will allow a fact to be associated with multiple criteria while being general enough, but not as encompassing as categories.

Commit:
Create tag.js component

Created a tag.js component using Chips from Material UI. We will use this component to implement tags that will display on our large and small fact cards. Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

Commit:
Change factCategory to factTags respectively

Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

Commit:
Change tags to a list 

Changed tags or category in config.yml to a list to be able to have multiple tags for one fact. Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

Commit:
Change categories to tags respectively

Changed wording of categories to tags in the respective files. Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu> Co-authored-by: Nicolas Robertson  <robe1606@morris.umn.edu>

Commit:
Change tags to gray background

Added a map to map over the list of tags in large-fact-card.js Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

UtkarshKr007 committed:
Change component name to Tag in tag.js

UtkarshKr007 committed:
Create a template to show facts grouped by a tag

Co-authored-by: NicolasRobertson <robe1606@morris.umn.edu>
This page is just a simple list for now, but it should show all facts containing a tag at a slug related to the tag. Next step is passing the tag as a context through gatsby-node.js

UtkarshKr007 commented:
Worked with @NicolasRobertson on the previous commit. Included the co-author in the commit, but he is not being credited for some reason.

UtkarshKr007 committed:
Configure gatsby-node.js to create page for tags

Also included the links to tags pages on homepage for now. Later, those
pages can be visited by clicking on the chips and possibly through the
sidebar.

UtkarshKr007 committed:
Add navigation functionality to chips

Now the chips can be clicked to navigate to their respective tag page.

Commit:
Add margin to tag style

UtkarshKr007 committed:
Merge branch 'master' into fact-Tags

UtkarshKr007 committed:
Change new facts 'category' field to 'tags' field

UtkarshKr007 committed:
Add tag above share button to compare

Commit:
Add cursor clickable functionality for tags

Added clickable-{true} to tag.js

Commit:
Move share button over to make room for tags

Commit:
Add media screen conditionally rendering tags

Added media screen queries that will make it so that on desktop it tags will be on the same line as the info button but on mobile tags will be on a different line so that there is enough room for multiple.

Commit:
Add spacing to tags for more visual appeal

Aligned tags to text above and added spacing if there was multiple tags to avoid overlap.

floogulinc commented on src/components/large-fact-card.js:
```diff
@@ -80,20 +93,37 @@ export default function LargeFactCard({factTitle, factImage, factSource, factHTM
+        <Grid item>
+          {factTags.map((factTag, index) => (
+            <Tag key={index} tagLabel={factTag.name} />
+          ))}
+        </Grid>
+      </Grid>
+
       <CardActions disableSpacing>
-        <Grid justify="space-between" container alignItems="flex-end">
+        <Grid justify="space-between" container alignItems="flex-end" wrap="nowrap">
+
+          <Grid item className={classes.inlineTags}>
+            {factTags.map((factTag, index) => (
+              <Tag key={index} tagLabel={factTag.name} />
+            ))}
+          </Grid>
+
           <Grid item>
-            {!isSSR && navigator.share &&
-              <Button color="primary" aria-label="share" onClick={shareClick}>
-                Share
-              </Button>
-            }
+          {!isSSR && navigator.share &&
+                <IconButton aria-label="share" onClick={shareClick}>
+                  <ShareIcon/>
+                </IconButton>
+              }
           </Grid>
-            <Grid item>
-              <IconButton aria-label="fact info" onClick={handleDialogClickOpen}>
-                <InfoOutlinedIcon />
-              </IconButton>
-            </Grid>
+
+          <Grid item>
+            <IconButton aria-label="fact info" onClick={handleDialogClickOpen}>
+              <InfoOutlinedIcon />
+            </IconButton>
+          </Grid>
```
This should all be in a single `<Grid item>` or we get this:

![image](https://user-images.githubusercontent.com/1300395/86565397-5d6bc980-bf2d-11ea-8827-d7e0d18c3a66.png)

floogulinc committed:
Tweak tag display on large fact card

floogulinc reviewed (changes requested):
I have made some tweaks that are too much to do in suggestions here so I made another branch and a PR to this branch: https://github.com/mn-pollinators/pollinator-facts/pull/111

floogulinc commented on src/components/tag.js:
```diff
@@ -0,0 +1,27 @@
+
+const tagStyles = makeStyles({
+  tag: {
+    backgroundColor: '#D3D3D3',
```
I don't think we need to set this slightly darker grey. The default is fine.

Commit:
Merge branch 'master' into fact-Tags

NicolasRobertson committed:
Create Layout for Tag Pages

Add toggle-layout to tagTemplate to improve tag page interface

Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

floogulinc committed:
Merge pull request #111 from mn-pollinators/fact-tags-tweaks

Tweak tag display on large fact card

floogulinc commented:
Example from today's discussion:
```ts
allPosts.filter(post => searchTags.every(st => post.tags.includes(st)))
```

floogulinc commented on src/components/large-fact-card.js:
```diff
@@ -28,12 +28,18 @@ const factCardStyles = makeStyles({
+   },
+  },
+  inlineTags: {
+    marginBottom: '8px',
```
```suggestion
    marginBottom: '12px',
```

Commit:
Move formatting from tag.js to large-fact-card

Added marginBottom 12px, removed  background color and moved formatting from tag component to large-fact-card.

Commit:
Fix styling so formatting refers to all elements in container

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

floogulinc reviewed (approved):
I think we might want to tweak what tags some of the facts have (eg should flowers get both `habitat` and `plant` (or maybe `flower`)) but other than that it looks good.
